workspace/document.go

```diff
@@ -0,0 +1,29 @@
+package workspace
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// Document is a file in a directory.
+type Document struct {
+	Root     string
+	Filepath string
+}
+
+// NewDocument creates a document from a filepath.
+func NewDocument(root, file string) Document {
+	return Document{
+		Root:     root,
+		Filepath: file,
+	}
+}
+
+// Path is the normalized path.
+// It uses forward slashes regardless of the operating system.
+func (doc Document) Path() string {
+	path := strings.Replace(doc.Filepath, doc.Root, "", 1)
```

Comment:
Are these two lines doing the same thing as `path, _ := filepath.Rel(doc.Root, doc.Filepath)`? Ignoring error right now.

Reply:
Oh. My. Godness.

I didn't know about `filepath.Rel`. That's _exactly_ what I need.